packages/did/src/Did.signature.ts

```diff
@@ -95,22 +104,64 @@ export async function verifyDidSignature({
       throw new SDKErrors.DidSubjectMismatchError(signer.did, expected.did)
     }
   }
+  if (signer.fragment === undefined) {
+    throw new SDKErrors.DidError(
+      `Signer DID URL "${signerUrl}" is not a valid DID resource.`
+    )
+  }
 
-  const { publicKey } = await didResolveKey(keyUri, expectedVerificationMethod)
+  const { contentStream, contentMetadata } = await dereferenceDidUrl(
+    signer.did,
+    {}
+  )
```

Comment:
The main problem with dereferencing the verification method, is that it does not tell you which relationships it has with the controller. I opened a PR in the DIF extensions repo: https://github.com/decentralized-identity/did-spec-extensions/pull/21, which will do just that. So, when this PR is merged, we can switch to dereferencing just the method, and requiring the method to be under a specific relationship.
Until then, as you mentioned, it is just simpler to resolve the full DID Document and parse its verification methods ourselves to find the one we are looking for. But again, since the current version of the resolver should be open for extensions, we can add this new logic once the PR is merged.

Reply:
That's an idea too! So far, as I mentioned, this is typically solved in a two-step process: resolve the verificationMethod first (which is the key), then resolve the controller document (i.e., DID document) based on the verificationMethod's controller. But this can certainly be optimised; you know you need the complete DID document eventually, so you might as well just load that and look up the key from it.